src/uthash.h

```diff
@@ -105,6 +105,22 @@ typedef unsigned char uint8_t;
 #define uthash_expand_fyi(tbl)            /* can be defined to log expands   */
 #endif
 
+/* whether we can recover from malloc() failure */
+#ifdef HASH_NOMEM_OK
+#undef HASH_NOMEM_OK
+#define HASH_NOMEM_OK 1
+#define SET_MEM_FAILED(hh,v) (hh)->mem_failed = (v)
+#define GET_MEM_FAILED(hh) ((hh)->mem_failed)
+#else
+#define HASH_NOMEM_OK 0
+#define SET_MEM_FAILED(hh,v) do{}while(0)
+#define GET_MEM_FAILED(hh) 0
+#endif
+
+#ifndef uthash_mem_failed
+#define uthash_mem_failed(elem) do{}while(0)
```

Comment:
This is the callback for memory failures. I thought you were adamant that it should exist (so the caller doesn't check the result mem_ok field). The parameter is the element that failed to add (or involved in memory failure in some way, but ATM, memory is allocated only when elements are being added).